engine/core/src/main/java/org/datacleaner/job/CompoundComponentRequirement.java

```diff
@@ -95,12 +130,21 @@ public boolean hasMultipleRequirementsFrom(HasFilterOutcomes producingComponent)
 
     @Override
     public boolean isSatisfied(InputRow row, FilterOutcomes outcomes) {
-        for (FilterOutcome outcome : outcomes.getOutcomes()) {
-            if (_outcomes.contains(outcome)) {
-                return true;
+        if(_compoundingType == CompundingType.ANY || row == null) {
```

Comment:
Note the `row == null`. Without this, the flow analysis went pretty crazy (ended up in the "This should never happen" Exception). This may have some serious drawbacks, but I'm not familiar enough with the flow analysis to be sure.